This is Solaris specific patch to use acceptable OpenSSL options by
default.

--- fetchmail-6.4.22/socket.c
+++ fetchmail-6.4.22/socket.c
@@ -926,7 +926,7 @@ static const char *SSLCertGetCN(const ch
 static int OSSL10X_proto_version_logic(int sock, const char **myproto, int *avoid_ssl_versions)
 {
 	if (!*myproto) {
-	    *myproto = "auto";
+	    *myproto = "TLS1.2+";
 	}
 
 	if (!strcasecmp("ssl3", *myproto)) {
@@ -1001,7 +1001,7 @@ static int OSSL110_proto_version_logic(i
 	SSL_CTX_set_min_proto_version(_ctx[sock], TLS1_VERSION);
 
 	if (!*myproto) {
-	    *myproto = "auto";
+	    *myproto = "TLS1.2+";
 	}
 
 	if (!strcasecmp("ssl3", *myproto)) {
@@ -1149,6 +1149,7 @@ int SSLOpen(int sock, char *mycert, char
 	}
 
 	SSL_CTX_set_options(_ctx[sock], sslopts | avoid_ssl_versions);
+	SSL_CTX_set_cipher_list(_ctx[sock], "ALL:!aNULL:!eNULL:!EXPORT:!LOW:!MD5:!DES:!3DES:!RC2:!RC4:!PSK:!kDH");
 
 	if (certck) {
 		SSL_CTX_set_verify(_ctx[sock], SSL_VERIFY_PEER, SSL_ck_verify_callback);
